File: baekjoon/1000~2999/2581_소수.c
#define _CRT_SECURE_NO_WARNINGS
#include <stdio.h>
#include <math.h>

int main(void) {

	int N, M;
	int i, j;
	int flag = 1;
	int arr[10000] = { 0 };
	int count = 0;
	int sum = 0;

	scanf("%d", &N);
	scanf("%d", &M);


	for (i = N; i <= M; i++) {
		flag = 0;

		if (i == 1)
			continue;
		for (j = 2; j <= sqrt(i); j++) {
			if (i%j == 0) {
				flag = 1;
				break;
			}
		}
		if (flag == 0) {
			arr[count++] = i;
		}
	}

	if (count == 0)
		printf("-1");

	else {
		for (int i = 0; i < count; i++)
			sum += arr[i];

		printf("%d %d", sum, arr[0]);
	}
	return 0;
}

File: baekjoon/1000~2999/2581_소수.py
from math import sqrt

arr = [0] * 10001
arr[0] = 1
arr[1] = 1
for i in range(2, int(sqrt(10001) + 1)):
    for j in range(i + i, 10001, i):
        if arr[j] == 0:
            arr[j] = 1

A = int(input())
B = int(input())

total = []
for i in range(A, B + 1):

    if arr[i] == 0:
        total.append(i)

if total:
    print(sum(total))
    print(min(total))

else:
    print(-1)